action.yml

```diff
@@ -44,12 +44,28 @@ outputs:
     description: 'Generated Docker tags'
   labels:
     description: 'Generated Docker labels'
+  annotations-index:
+    description: 'Generated index level annotations'
+  annotations-index-descriptor:
+    description: 'Generated index descriptor level annotations'
+  annotations-manifest:
+    description: 'Generated manifest level annotations'
+  annotations-manifest-descriptor:
+    description: 'Generated manifest descriptor level annotations'
   json:
     description: 'JSON output of tags and labels'
   bake-file-tags:
     description: 'Bake definition file with tags'
   bake-file-labels:
     description: 'Bake definition file with labels'
+  bake-file-annotations-index:
```

Comment:
This 4 separate files is kind of weird in here. How would the user know which is the correct one to use? It looks like we don't know which one is correct and are pushing for the user to make the mistake for us.

Maybe this makes more sense with:
- there is only one annotation file
- we pick what is the correct default
- there is some optional override to choose non-default (or multiple?) if user really wants to mess it up.

Reply:
Yes we could provide a `bake-file-annotations` output that would default to annotations at manifest level like BuildKit does as default https://github.com/moby/buildkit/blob/master/docs/annotations.md#image-annotations

> there is some optional override to choose non-default (or multiple?) if user really wants to mess it up.

We could alter the `bake-file-annotations` using an env var like `DOCKER_METADATA_ANNOTATIONS_FORMAT` that would default to `manifest` and user could pass multiple values like `DOCKER_METADATA_ANNOTATIONS_FORMAT=manifest,index`:

```yaml
      -
        name: Docker meta
        uses: docker/metadata-action@v5
        with:
          images: name/app
      -
        name: Build
        uses: docker/bake-action@v3
        with:
          files: |
            ./docker-bake.hcl
            ${{ steps.meta.outputs.bake-file-tags }}
            ${{ steps.meta.outputs.bake-file-annotations }}
          targets: build
        env:
          DOCKER_METADATA_ANNOTATIONS_FORMAT: manifest,index
```

WDYT?

Reply:
Updated: https://github.com/moby/buildkit/blob/master/docs/annotations.md